### PR TITLE
Make empty results ok

### PR DESCRIPTION
**Pull request by jdblack**
Carbonserver currently returns 400 if the result set is empty.   Graphite-web handles this result poorly,  resulting in graphite-web returning 500 series errors to grafana.    This  PR  provides an option in go-carbon which  configures gocarbon to return a 200 result if the result set is empty.

### CODE REVIEW

**Comment by jdblack** [comment]
The DeepSource  failing appears to be from unrelated code.

File: carbonserver/render.go [hunk 1]
```diff
@@ -221,7 +221,7 @@ func (listener *CarbonserverListener) renderHandler(wr http.ResponseWriter, req
 		return
 	}
 
-	if response.metricsFetched == 0 {
+	if response.metricsFetched == 0 && listener.emptyResultOk == false  {
```
Comment (deniszh) [inline carbonserver/render.go]: Second Deepsoure warning has sense. `&& listener.emptyResultOk == false` should be better replaced with `&& !listener.emptyResultOk`

**Comment by deniszh** [comment]
@jdblack : could you also please apply go fmt ? linter also complains - https://github.com/go-graphite/go-carbon/runs/5535152765?check_suite_focus=true

**Comment by deniszh** [comment]
@jdblack : OT : and I'm joining to @bom-d-van question too - are you really running Graphite cluster with 3B metrics, as you mentioned in https://github.com/go-graphite/go-carbon/issues/424#issuecomment-1064398730 ???

**Comment by jdblack** [comment]
Thanks for the PR review!   I fixed the conditional as asked.  I ran  "gofmt -w -s" on the golang files that I touched, and none seemed modified.   A  "gofmt -w -s ."  showed changes to 4 unrelated files, mostly related to the trie code.  Would you like me to gofmt  those? 

Yes, in our biggest environment  I do mean to answer @bom-d-van,  but time  got away from me on Friday. I promise I'll put something together early this week,  but here's a pic =)  https://imagebin.ca/v/6aCg0IgCamRF  .

**Comment by deniszh** [comment]
@jdblack : now everything looks good, thanks!
LGTM, could you also review, @bom-d-van  ?

**Review by deniszh — APPROVED** [review]

**Review by bom-d-van — APPROVED** [review]
hi, lgtm as well. the deepsource complain isn't introduced in this pr. so I guess we can skip it. (will open a ticket to track it, seems a reasonable optimisation suggestion). thanks for the new feature.

---

> Yes, in our biggest environment I do mean to answer @bom-d-van, but time got away from me on Friday. I promise I'll put something together early this week, but here's a pic =) https://imagebin.ca/v/6aCg0IgCamRF .

wow, 3 billions... that's nuts. 👍 

image

**Pull request merged**